getgist/github.py

```diff
@@ -42,14 +42,20 @@ def __init__(self, user, file_path, assume_yes=False):
         Save basic variables to all methods, instantiate GetGistrequests and
         calls the OAuth method.
         :param user: (str) GitHub username
-        :param file_path: (str) file_path to be saved (locally), created or
-        updated (remotelly)
+        :param file_path: (str or None) file_path to be saved (locally), created or
+        updated (remotelly) 
         :param assume_yes: (bool) assume yes (or first option) for all prompts
         :return: (None)
         """
         self.user = user
         self.file_path = file_path
-        self.filename = os.path.basename(file_path)
+        
+        if self.file_path:
+            self.filename = os.path.basename(file_path)
+
+        else:
+            self.filename = None
+
```

Comment:
I think these blank lines make readability worse: it looks we are moving on to different logical parts. Also, we could have a simpler expression:

```suggestion
        self.filename = os.path.basename(file_path) if file_path else None
```